.github/workflows/update_sdk_methods.py

```diff
@@ -445,6 +452,50 @@ def make_soup(url):
    except urllib.error.HTTPError as err:
        print(f'An HTTPError was thrown: {err.code} {err.reason} for URL: {url}')
 
+## Link any matching data types to their reference links, based on {sdk}_datatype_links[] array,
+## used in parse() for both param and return data types. Handles data types syntax that includes
+## multiple data types (and therefore requires multiple data type links), such as
+## ListOrganizationMembers: Tuple[List[viam.proto.app.OrganizationMember], List[viam.proto.app.OrganizationInvite]
+## DESIGN DECISION: Ignore well-known, usually leading (containing) data types like List, Tuple, Dict.
+## NOTE: Only used in PySDK parsing, for now (but should work for all with minor tweak to support per-language links array):
+def link_data_types(sdk, data_type_string):
+
+    linked_data_type_string = ""
+
+    ## If the passed data_type_string matches exactly to a data type defined in python_datatype_links, use that:
+    if data_type_string in python_datatype_links.keys():
+       linked_data_type_string = '[' + data_type_string + '](' + python_datatype_links[data_type_string] + ')'
+    else:
+
+        ## Assemble all encountered data types that match to python_datatype_links keys into array.
+        ## This match is a little too greedy, and will match, say, 'int' to 'JointPositions'. To counter
+        ## this, we additionally check for leading and trailing alphanumeric characters further in:
+        matching_data_types = list(key for key in python_datatype_links if key in data_type_string)
+
+        if len(matching_data_types) > 0:
+
+            ## Ugly hack to allow us to append within the for loop below, sorry:
+            linked_data_type_string = data_type_string
+
+            for data_type_found in matching_data_types:
+
+                ## Discard string matches that are substrings of other data type strings:
+                if not regex.search(r'[A-Za-z0-9]' + data_type_found, data_type_string) and not regex.search(data_type_found + r'[A-Za-z0-9]', data_type_string):
+
+                    data_type_linked = '[' + data_type_found + '](' + python_datatype_links[data_type_found] + ')'
+                    linked_data_type_string = regex.sub(data_type_found, data_type_linked, linked_data_type_string)
+                else:
+                    ## If we get here, this data_type is actually a substring of another data type. Take no action:
+                    pass
+
+    ## If we didn't find any matching links, return an empty string so we can know to look elsewhere,
+    ## otherwise return linked data type string:
+    if linked_data_type_string == data_type_string:
+        return ""
```

Comment:
Wouldn't we want to return something that still produced a readable result and just not have a link? So if the sdk has an int but we don't know the link for it, we'll gave int in our docs just without link?
```suggestion
        return data_type_string
```

potentially we'd want to add logging I suppose

Reply:
nvm - I see where it's used now